### DISCUSSION TITLE
Support Bitcoin Taproot (Generating a Taproot Address)

### DISCUSSION THREAD

**Discussion by hitripod** [trustwallet/wallet-core#3155] [Ideas & Feature request]
Sending to Taproot address is mostly covered in wallet-core, but we should support Taproot as the default address, incl. receiving to Taproot address.

Sending to Taproot address is mostly covered in wallet-core, but I'm not yet 100% certain.

**Is your feature request related to a problem? Please describe.**

https://github.com/trustwallet/wallet-core/issues/1785

**Describe the solution you'd like**

Support taproot: https://github.com/trustwallet/wallet-core/blob/master/registry.json

**Comment by CryptoManiac** [comment]
Unless you need to tweak the public key for some reason, not much actually needed to be done. With that in mind, P2TR essentially becomes a reincarnation of P2PK. Payments are done to prefixed public key and the witness data only consists of the signature.

**Reply by CryptoManiac** [reply]
> Unless you need to tweak the public key for some reason, not much actually needed to be done. With that in mind, P2TR essentially becomes a reincarnation of P2PK. Payments are done to prefixed public key and the witness data only consists of the signature.

In fact I did exactly that yesterday: https://bb-btc.mycelium.com:19130/tx/71fbd171e3af71d1547e98fcd7504581899274b7b8e7905dff49c3e8ff00d5a6

I forked this repo, added libsecp256k1 dependency and implemented taproot sighash function. Adding new branch to sigStep function was a trivial matter after that. Not going to publish my code though. Not anytime soon, at least, as it is too sensitive in terms of money and I was very drunk while writing it.

**Comment by lamafab** [comment]
We just merged this major PR: https://github.com/trustwallet/wallet-core/pull/3233

We support Taproot generation internally, but don't expose it via the C interface yet (BRC20 was prioritized). Working on that next.

**Reply by dinghaoz** [reply]
When taproot interface will be supported?
